dotnet/src/Connectors/Connectors.Ollama/Extensions/OllamaServiceCollectionExtensions.cs

```diff
@@ -135,10 +147,24 @@ public static IServiceCollection AddOllamaChatCompletion(
         Verify.NotNull(services);
 
         services.AddKeyedSingleton<IChatCompletionService>(serviceId, (serviceProvider, _) =>
-            new OllamaChatCompletionService(
-                modelId: modelId,
-                httpClient: HttpClientProvider.GetHttpClient(httpClient, serviceProvider),
-                loggerFactory: serviceProvider.GetService<ILoggerFactory>()));
+        {
+            var ollamaClient = new OllamaApiClient(
+                client: HttpClientProvider.GetHttpClient(httpClient, serviceProvider),
+                modelId);
+
+            var logger = serviceProvider.GetService<ILoggerFactory>()?.CreateLogger(ollamaClient.GetType());
+
+            var chatClientBuilder = new ChatClientBuilder()
+                .UseFunctionInvocation(config =>
+                    config.MaximumIterationsPerRequest = 128);
+
+            if (logger is not null)
```

Comment:
nitpicking: creating the logger (line 155), could be moved to "just" before it is being used (before the `if (logger is not null)`